src/index.ts

```diff
@@ -123,7 +123,9 @@ export interface TableMetadata {
 
 export interface FileUploadOptionsSpec {
   type: UploadType;
-  data: string | File;
+  data: File;
+  url: string;
```

Comment:
To add onto this, I think we should split `FileUploadOptionsSpec` into two interfaces, one for tables and one for networks. They would look like so:

```typescript
export interface TableUploadOptionsSpec {
  data: File;
  edgeTable?: boolean;
  key?: string;
  overwrite?: boolean;
  columnTypes?: {
    [key: string]: ColumnType;
  };
}

export interface NetworkUploadOptionsSpec {
  data: File;
  type: GraphUploadType;
}
```

While I think there's other long-term improvements we can make here, these one are in scope and would make things more clear. Thoughts @eagw @waxlamp?

Reply:
That makes sense.  If we split it up, I would actually make `edgeTable` and `columnTypes` required, since they were only optional in `FileUploadOptionsSpec` because you need them when you upload a table, but not for a network. My only question would be, since you [suggested](https://github.com/multinet-app/multinetjs/pull/44#discussion_r687246691) not passing `overwrite` and `key` in the call anymore @AlmightyYakob, do you think we still need them here, even if they're optional?

Reply:
> My only question would be, since you [suggested](https://github.com/multinet-app/multinetjs/pull/44#discussion_r687246691) not passing `overwrite` and `key` in the call anymore @AlmightyYakob, do you think we still need them here, even if they're optional?

My bad. I think it's fine to remove `overwrite` and `key` as well, although we'll probably want them back once we add api support for those fields. I'll file an issue to track that, but I think for now removing those fields makes sense.


> If we split it up, I would actually make `edgeTable` and `columnTypes` required, since they were only optional in `FileUploadOptionsSpec` because you need them when you upload a table, but not for a network.

I think `edgeTable` should definitely be required, but `columnTypes` is actually still optional in the [API](https://github.com/multinet-app/multinet-api/blob/1d5c6e731e249f6176dbfd8141883e15a17f49e9/multinet/api/views/serializers.py#L161-L164) (notice the `default=dict`).

Reply:
> My bad. I think it's fine to remove `overwrite` and `key` as well, although we'll probably want them back once we add api support for those fields. I'll file an issue to track that, but I think for now removing those fields makes sense.

No problem!  I wasn't sure either way.
> I think `edgeTable` should definitely be required, but `columnTypes` is actually still optional in the [API](https://github.com/multinet-app/multinet-api/blob/1d5c6e731e249f6176dbfd8141883e15a17f49e9/multinet/api/views/serializers.py#L161-L164) (notice the `default=dict`).

Ah, that's right. Thanks!

Reply:
(Just to answer the question put to me: splitting up the options spec this way is good and necessary IMO.)

Reply:
They should be good to go.

See 846d1f3